src/compiler/checker.ts

```diff
@@ -13916,14 +13916,10 @@ namespace ts {
 
                 checkDecorators(node);
                 checkSignatureDeclaration(node);
+                // TODO (weswig): Consolidate the below into checkAllCodePathsInNonVoidFunctionReturnOrThrow

```

Comment:
Mmmm... Maybe. I wanted a second opinion on that. The remaining code below
is getter-specific, so I'm unsure of weather or not it should be merged
into checkAllCodePathsInNonVoidFunction.

On Mon, Aug 8, 2016, 11:01 AM Anders Hejlsberg <notifications@github.com>
wrote:

> In src/compiler/checker.ts
> <https://github.com/Microsoft/TypeScript/pull/10102#discussion_r73922852>:
>
> > @@ -13916,14 +13916,10 @@ namespace ts {
> >
> >                  checkDecorators(node);
> >                  checkSignatureDeclaration(node);
> > +                // TODO (weswig): Consolidate the below into checkAllCodePathsInNonVoidFunctionReturnOrThrow
>
> Does this imply there's more work to do?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/Microsoft/TypeScript/pull/10102/files/89419511803e3c3e8cae0ccad4a9678d272b199e#r73922852>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ACzAMihmhfYNeSrrNmgI6Y63i2Cf1cZWks5qd27jgaJpZM4JbGAe>
> .
>